packages/paypal-commerce-integration/src/paypal-commerce-alternative-methods/paypal-commerce-alternative-method-ratepay-payment-strategy.ts

```diff
@@ -0,0 +1,233 @@
+
+        return `${date.getFullYear()}-${formattedMonth}-${formattedDate}`;
+    }
+
+    private renderLegalText() {
+        if (this.paypalcommerceratepay) {
+            const legalTextContainerId = this.paypalcommerceratepay.legalTextContainer;
+            const { container } = this.paypalcommerceratepay;
+            const buttonContainerId = container.split('#')[1];
+            const buttonContainer = document.getElementById(buttonContainerId);
+            const buttonContainerParent = buttonContainer?.parentNode;
+            const legalTextContainer = document.createElement('div');
+            legalTextContainer.style.marginBottom = '20px';
+            legalTextContainer.setAttribute('id', legalTextContainerId);
+            buttonContainerParent?.prepend(legalTextContainer);
+            const paypalSdk = this.paypalCommerceIntegrationService.getPayPalSdkOrThrow();
+            const ratePayButton = paypalSdk.Legal({
+                fundingSource: paypalSdk.Legal.FUNDING.PAY_UPON_INVOICE,
+            });
+            const legalTextContainerElement = document.getElementById(legalTextContainerId);
+
+            if (legalTextContainerElement) {
+                ratePayButton.render(`#${legalTextContainerId}`);
+            }
+        }
+    }
+    private handleError(error: unknown): void {
+        const { onError } = this.paypalcommerceratepay || {};
+        this.toggleLoadingIndicator(false);
+
+        if (onError && typeof onError === 'function') {
+            onError(error);
+        }
+    }
+
+    private createFraudNetScript(merchantId: string, methodId: string, gatewayId: string) {
+        const state = this.paymentIntegrationService.getState();
+        const paymentMethod = state.getPaymentMethodOrThrow<PayPalCommerceInitializationData>(
+            methodId,
+            gatewayId,
+        );
+        const { testMode } = paymentMethod.config;
+        const scriptElement = document.createElement('script');
+        scriptElement.setAttribute('type', 'application/json');
+        scriptElement.setAttribute('fncls', 'fnparams-dede7cc5-15fd-4c75-a9f4-36c430ee3a99');
+        this.guid = this.generateGUID();
+        const jsonObject = {
+            f: this.guid,
+            s: `${merchantId}_checkout-page`,
+            sandbox: testMode,
+        };
+
+        scriptElement.innerHTML = JSON.stringify(jsonObject);
+
+        document.body.appendChild(scriptElement);
+    }
+
+    private loadFraudnetConfig() {
+        const script = document.createElement('script');
+        script.src = 'https://c.paypal.com/da/r/fb.js';
+        document.body.appendChild(script);
+    }
+
+    private generateGUID() {
+        const chars = 'abcdef0123456789';
+        let guid = '';
+
+        for (let i = 0; i < 32; i + 1) {
+            const index = Math.floor(Math.random() * chars.length);
+            guid += chars.charAt(index);
+        }
+
+        return guid;
+    }
```

Comment:
This method can be removed, because you can use a specific method from lodash that @bc-nick mentioned before

Reply:
replaced with lodash uniqueId

Reply:
Returned to generateGuid func because lodash gives us uniqId but it doesn't match our requirements